### PR TITLE
Update Post Caption

### PR DESCRIPTION
**Pull request by abhishek3434**
**Solved-Update feature for Posts #151**
- Added a dialog box from MaterialUI to edit post caption

Screenshot from 2023-05-21 03-15-41
Screenshot from 2023-05-21 03-15-52
Screenshot from 2023-05-21 03-15-02

### CODE REVIEW

File: src/components/Post.jsx [hunk 1]
```diff
@@ -45,8 +46,10 @@ function Post(prop) {
   const [comment, setComment] = useState("");
   const [likesNo, setLikesNo] = useState(likecount ? likecount.length : 0);
   const [anchorEl, setAnchorEl] = useState(null);
+  const[editCaption,setEditCaption]=useState(caption)
```
Comment (narayan954) [inline src/components/Post.jsx:49]: please maintain a proper format, try using auto formatter like prettier to format the code :)

File: src/components/Post.jsx [hunk 2]
```diff
@@ -45,8 +46,10 @@ function Post(prop) {
...
   const [showEmojis, setShowEmojis] = useState(false);
   const [Open, setOpen] = useState(false);
+  const [openEditCaption,setOpenEditCaption]=useState(false);
```
Comment (narayan954) [inline src/components/Post.jsx:52]: again formatting improper

File: src/components/Post.jsx [hunk 3]
```diff
@@ -184,6 +187,28 @@ function Post(prop) {
     setOpen(true);
     setAnchorEl(null);
   };
+  const handleClickOpenCaption=async()=>{
+
```
Comment (narayan954) [inline src/components/Post.jsx:191]: extra line, lots of formatting issues. format the whole code please using a formatter

File: src/components/Post.jsx [hunk 4]
```diff
@@ -184,6 +187,28 @@ function Post(prop) {
...
+    
+    setOpenEditCaption(true);
+  }
+  const handleClickClosedCaption=()=>{
+    setEditCaption(caption)
+    setOpenEditCaption(false);
+  }
+  const handleSubmitCaption=async()=>{
+    const taskDocRef = doc(db, "posts", postId);
+    let data= await db.collection("posts").doc(postId).get().then(doc =>{
```
Comment (narayan954) [inline src/components/Post.jsx:200]: was this query even needed??

**Review by narayan954 — CHANGES_REQUESTED** [review]
please apply code formatting and remove unnecessary elements as asked

**Comment by abhishek3434** [comment]
> please apply code formatting and remove unnecessary elements as asked

Completed please check now

File: src/components/Post.jsx (hunk 4 above)
Comment (abhishek3434) [reply src/components/Post.jsx]: No need of this query removed noq

File: src/components/Post.jsx [hunk 5]
```diff
@@ -1,3 +1,13 @@
+import { useEffect, useState } from "react";
```
Comment (narayan954) [inline src/components/Post.jsx:1]: why is there duplicate import of modules? please fix that and test if the application runs without issue.

**Comment by narayan954** [comment]
Also, please try to solve the merge conflict

**Pull request closed without merging**